inc/session_handler/class-wp-cache-session-handler.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Altis Cloud Session handler.
+ *
+ * @package altis/cloud
+ */
+
+namespace Altis\Cloud\Session_Handler;
+
+use SessionHandlerInterface;
+
+/**
+ * WP Cache Session Handler.
+ *
+ * Uses the WordPress object cache to store and retrieve session data.
+ */
+class WP_Cache_Session_Handler implements SessionHandlerInterface {
+	protected int $ttl;
+
+	/**
+	 * Initialize the session handler.
+	 */
+	public function __construct() {
+		wp_cache_add_global_groups( 'sessions' );
+		$this->ttl = ini_get( 'session.gc_maxlifetime' );
+	}
+
+	/**
+	 * Initialize a session.
+	 *
+	 * No-op in this session handler as there is nothing to
+	 * initialize, the object cache takes care of all that.
+	 *
+	 * @param string $path
+	 * @param string $name
+	 *
+	 * @return bool
+	 */
+	public function open( string $path, string $name ) : bool {
+		return true;
+	}
+
+	/**
+	 * Close the session.
+	 *
+	 * No-op as there is nothing to close.
+	 *
+	 * @return bool
+	 */
+	public function close() : bool {
+		return true;
+	}
+
+	/**
+	 * Destroy a session.
+	 *
+	 * @param string $id
+	 *
+	 * @return bool
+	 */
+	public function destroy( string $id ) : bool {
+		return wp_cache_delete( $id, 'sessions' );
+	}
+
+	/**
+	 * Read session data.
+	 *
+	 * @param string $id
+	 *
+	 * @return string|false
+	 */
+	public function read( string $id ) : string | false {
+		$data = wp_cache_get( $id, 'sessions' );
+		if ( ! $data ) {
```

Comment:
This may want to use `$found`? or I guess if it's only ever a string, it should be safe enough?

Reply:
Yeah, it throws a warning if you return anything that's not a string (probably because `open()` succeeded), so even for a session that was not found we have to return an empty string. We know `wp_cache_get()` will return `false` if the value doesn't exist, because `$data` is a `string` in `write()` but that still should be returned as an empty string to avoid the warning.

I believe this is because `open()` is typically responsible for determining whether the session exists, i.e. tries to open a file descriptor, so when a descriptor is open, `read()` is almost guaranteed to succeed. Maybe moving `wp_cache_get()` to `open()`, storing the value in a property and then returning it in `read()`? Maybe that's more semantically accurate, but that makes things more confusing, especially given that the Redis connection is abstracted anyway. Forcing a string is a good tradeoff I think.